### PR TITLE
Feature/x quest

### PR DESCRIPTION
**Pull request by timosachsenberg**

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -45,3 +45,5 @@ to best get started:
   see http://openms.de/support for up-to-date information
   about how to contact the OpenMS developers.
 
+dies ist ein test
+test test test
```
Comment (hroest) [inline README]: is this currently in a shape to be merged or are you still working on it?

File: src/utils/xQuest.cpp [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+#ifdef _OPENMP
+#pragma omp atomic
+#endif
+        ++count_peptides;
+
+
+        vector<AASequence> all_modified_peptides;
+
+        // generate all modified variants of a peptide
+        // Note: no critial section is needed despite ResidueDB not beeing thread sage.
```
Comment (hroest) [inline src/utils/xQuest.cpp]: - critical
- safe

File: src/utils/xQuest.cpp [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+        //       It is only written to on introduction of novel modified residues. These residues have been already added above (single thread context).
```
Comment (hroest) [inline src/utils/xQuest.cpp]: wouldnt it anyways make sense to add thread safety to Residue DB ? did we discuss this previously, does it impact performance too much ?

File: src/utils/xQuest.cpp [hunk 4]
```diff
@@ -0,0 +1,589 @@
...
+            const Size scan_index_heavy = scan_index_light_it->second;
+
+            cout << "light spectrum index: " << scan_index_light << " heavy spectrum index: " << scan_index_heavy << endl;
+            const PeakSpectrum& spectrum_heavy = spectra[scan_index_heavy];
+
+            // TODO: common peaks, align
+          }
+          
+
+//          HyperScore::IndexScorePair best_score = HyperScore::compute(fragment_mass_tolerance, fragment_mass_tolerance_unit_ppm, exp_spectrum, theoretical_spectra);
```
Comment (hroest) [inline src/utils/xQuest.cpp]: is there some part missing, e.g. the part that computes the hyper score?

**Comment by timosachsenberg** [comment]
Hi @hroest. Sorry somehow it seems this PR showed up accidentally. I closed it about 3 hours ago. It is in a very early stage (actually only one day of hacking with a new PhD student). Thanks for pointing out some issues.

File: src/utils/xQuest.cpp (hunk 3 above)
Comment (timosachsenberg) [inline src/utils/xQuest.cpp]: I think we concluded that a "simple" solution is to implement a reader(s) writer lock. Seems not that straightforward but doable.

**Comment by timosachsenberg** [comment]
It currently also depends on an unmerged branch of mine for the RNA protein cross-linking.

**Pull request closed without merging**